Add stddev/var functions (#877)
risingwavelabs/risingwave-docs: sql/functions/json.mdx

[2, false]
```

Note: If you're using `jsonb_array_elements` to do array unnesting for row-level processing, followed by an `array_agg` aggregation,
it can incur additional overhead due to subplan execution. Use `EXPLAIN` to examine your query plan, and see if there are unexpected `JOIN`s in your query.

> If so, please consider rewriting with `array_transform` to perform your mapping logic to avoid the subplan execution.

Reviewer: ```suggestion
<Note>
If you're using `jsonb_array_elements` to do array unnesting for row-level processing, followed by an `array_agg` aggregation,
it can incur additional overhead due to subplan execution. Use `EXPLAIN` to examine your query plan, and see if there are unexpected `JOIN`s in your query.
If so, please consider rewriting with `array_transform` to perform your mapping logic to avoid the subplan execution.
</Note>
```

[2, false]
```

<Note>
If you're using `jsonb_array_elements` to do array unnesting for row-level processing, followed by an `array_agg` aggregation,
it can incur additional overhead due to subplan execution. Use `EXPLAIN` to examine your query plan, and see if there are unexpected `JOIN`s in your query.

> If so, please consider rewriting with `array_transform` to perform your mapping logic to avoid the subplan execution.

Reviewer 2: ```suggestion
If you're using `jsonb_array_elements` to unnest arrays for row-level processing and then apply an `array_agg` aggregation,
it can incur additional overhead due to subplan execution. Use `EXPLAIN` to examine your query plan and check for unexpected JOINs.
If such joins appear, consider rewriting the query using `array_transform` to perform your mapping logic and avoid subplan execution.
```

</Note>

For example, given this original SQL:
```sql
create table t(key int, payload jsonb);
explain create materialized view m1 as 
  select payload->>'id' as id,

>   (select array_agg(value) from (select jsonb_array_elements(payload->'elements') as value)) as elements

Reviewer 3: ```suggestion
  (select array_agg(value->>'subField' order by ordinality) from jsonb_array_elements(payload->'elements') with ordinality as value) as elements
```

* Without `->>'subField'` there is no need to `array_transform` - it is a pure `jsonb_to_array`
* Without `order by ordinality` this query may not preserve element order

Reviewer 3: It is not enough to update the SQL alone. The example plan below is now out of sync.

Reviewer: Fixed, my bad.

           ├─StreamProject { exprs: [t.payload] }
           │ └─StreamHashAgg { group_key: [t.payload], aggs: [count] }
           │   └─StreamExchange { dist: HashShard(t.payload) }
           │     └─StreamTableScan { table: t, columns: [payload, _row_id] }
           └─StreamProject { exprs: [t.payload, JsonbAccessStr(JsonbArrayElements(JsonbAccess($0, 'elements':Varchar)), 'subField':Varchar) as $expr1, (projected_row_id + 1:Int64) as $expr2, 1:Int32, projected_row_id] }
             └─StreamProjectSet { select_list: [$0, JsonbArrayElements(JsonbAccess($0, 'elements':Varchar))] }
               └─StreamProject { exprs: [t.payload] }
                 └─StreamHashAgg { group_key: [t.payload], aggs: [count] }
                   └─StreamExchange { dist: HashShard(t.payload) }
                     └─StreamTableScan { table: t, columns: [payload, _row_id] }
(18 rows)
```

If we rewrite it with array_transform, the plan is greatly simplified:
```sql
explain create materialized view m1 as 
  select payload->>'id' as id,
  array_transform(jsonb_to_array(payload->'elements'), |x| x->>'subField') as elements

> from t;

Reviewer: This shouldn't need `WITH ORDINALITY` right? Since we don't unnest, but just extract the entire array. Further array_transform will do a stable transform, and not move the position of elements as well.

Reviewer: @xiangjinwu

Reviewer 3: Right
